examples/get-post/src/get_post/core.clj

```diff
@@ -0,0 +1,84 @@
+(ns get-post.core
+  (:require [org.httpkit.server :as server]
+            [clj-http.client :as client]
+            [plumbing.core :refer [defnk]]
+            [kekkonen.cqrs :as cqrs]
+            [kekkonen.core :as k]
+            [schema.core :as s]))
+
+(s/defschema GetInput
+             {:name                         s/Str
+              (s/optional-key :description) s/Str})
+
+; GET http://localhost:3000/api/get-and-post?name=taras
+(s/defn get-handler
+        "Echoes a GetInput"
+        [data :- GetInput]
+        ; here is your handler
+        (cqrs/success data))
+
+(s/defschema PostInput
+             {:data s/Str})
+
+; POST http://localhost:3000/api/get-and-post {"data":"taras"}
+(s/defn post-handler
+        "Echoes a PostInput"
+        [data :- PostInput]
+        ; here is your handler
+        (cqrs/success data))
+
+(defnk ^:query simple-get
+       "handles get"
+       [data request]
+       (cqrs/success data))
+
+(defnk ^:get-post get-and-post
+       "handles both requests"
+       [get-params post-params request]
+       (if (= (:request-method request) :get)
+         (s/with-fn-validation (get-handler get-params))
```

Comment:
there is `kekkonen.core/input-coerce!` that can be used for this => uses the registered http-coercions and produces identical errors than the standard coercion.